### ISSUE TITLE
`line-too-long` (`E501`) - false positive on import statements from long module names where it's impossible to fix

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#15236]
if a line contains an import statement from a very long module path, it triggers `E501`, even though there's no way to fix it as far as i know:

```py
from asdfasasd.asdfasdf.asdfasdfsa.dfasdfasdfasdf.asdfasdfasdf.asdfasdfasdf.asdfasdfasdf.asdfsasdfasdf import ( # error: E501
    foo,
)
```

https://play.ruff.rs/93d32aef-cefb-41f2-8205-e05bbea8ba28?secondary=Format

ruff should be able to determine that this situation is unavoidable and therefore should not report the error. it already does this in some cases for example comments/docstrings with long URLs

**Comment by bramth** [comment]
As far as I am aware, this has not yet been picked up. Any reasons for not doing so? Often contracts with for example protobuf files fail this requirement.
